File: hw/ramos/u3/hw35StrategyPersistance/StrategyPersistencePY/controller/MongoStrategy.py
from pymongo import MongoClient
from controller.PersistenceStrategy import PersistenceStrategy
from model.Store import Store

class MongoStrategy(PersistenceStrategy):
    def __init__(self):
        self.client = MongoClient("mongodb+srv://Paulo:paulo2004@cluster0.9uxqgih.mongodb.net/")
        self.db = self.client["Store"]
        self.col = self.db["Store"]

    def create(self, s):
        self.col.insert_one(s.to_dict())

    def find(self, id):
        doc = self.col.find_one({"id": id})
        return Store.from_dict(doc) if doc else None

    def update(self, id, s):
        self.col.replace_one({"id": id}, s.to_dict())

    def delete(self, id):
        self.col.delete_one({"id": id})

    def load_all(self):
        return [Store.from_dict(doc) for doc in self.col.find()]

File: hw/ramos/u3/hw35StrategyPersistance/StrategyPersistence/src/main/java/ec/edu/espe/strategypersistence/controller/MongoStrategy.java
package ec.edu.espe.strategypersistence.controller;

import com.mongodb.client.*;
import static com.mongodb.client.model.Filters.eq;
import ec.edu.espe.strategypersistence.model.Store;
import org.bson.Document;
import java.util.ArrayList;
import java.util.List;


/**
 *
 * @author Paulo Ramos
 */

public class MongoStrategy implements PersistenceStrategy {
    
    private MongoCollection<Document> collection;

    public MongoStrategy() {
        MongoClient client = MongoClients.create("mongodb+srv://Paulo:paulo2004@cluster0.9uxqgih.mongodb.net/");
        MongoDatabase db = client.getDatabase("Store");
        this.collection = db.getCollection("Store");
    }

    @Override
    public void create(Store s) {
        Document doc = new Document("id", s.getId())
                .append("name", s.getName())
                .append("price", s.getPrice())
                .append("category", s.getCategory());
        collection.insertOne(doc);
    }

    @Override
    public Store find(int id) {
        Document doc = collection.find(eq("id", id)).first();
        return doc == null ? null : documentToStore(doc);
    }

    @Override
    public void update(int id, Store s) {
        Document doc = new Document("id", s.getId())
                .append("name", s.getName())
                .append("price", s.getPrice())
                .append("category", s.getCategory());
        collection.replaceOne(eq("id", id), doc);
    }

    @Override
    public void delete(int id) {
        collection.deleteOne(eq("id", id));
    }

    @Override
    public List<Store> loadAll() {
        List<Store> list = new ArrayList<>();
        for (Document doc : collection.find()) {
            list.add(documentToStore(doc));
        }
        return list;
    }

    private Store documentToStore(Document doc) {
        return new Store(doc.getInteger("id"), doc.getString("name"), 
                         doc.getDouble("price").floatValue(), doc.getString("category"));
    }
}